core/src/epoch_slots.rs

```diff
@@ -1,24 +1,52 @@
 use crate::cluster_info::MAX_CRDS_OBJECT_SIZE;
+use crate::crds_value::MAX_SLOT;
+use crate::crds_value::MAX_WALLCLOCK;
 use bincode::serialized_size;
 use bv::BitVec;
 use flate2::{Compress, Compression, Decompress, FlushCompress, FlushDecompress};
 use solana_sdk::clock::Slot;
 use solana_sdk::pubkey::Pubkey;
+use solana_sdk::sanitize::{Sanitize, SanitizeError};
 
+const MAX_SLOTS_PER_ENTRY: usize = 2048 * 8;
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 pub struct Uncompressed {
     pub first_slot: Slot,
     pub num: usize,
     pub slots: BitVec<u8>,
 }
 
+impl Sanitize for Uncompressed {
+    fn sanitize(&self) -> std::result::Result<(), SanitizeError> {
+        if self.first_slot >= MAX_SLOT {
+            return Err(SanitizeError::ValueOutOfRange);
+        }
+        if self.num >= MAX_SLOTS_PER_ENTRY {
```

Comment:
is this check valid? `self.num < self.slots.len()`

Reply:
yea, it's redundant, but i want the wire sanitize stuff to have low cognitive overhead

Reply:
It's being done now ad-hoc in the implementation right? Seems lower cognitive overhead to have it one place and less risk if someone changes the implementation logic in the future without coding for possible large values.

Reply:
The sanitize thing is a separate trait, vs the index check in the implementation.  `MAX_SLOTS_PER_ENTRY` is by design, because the object should compress into a smallish entry that can be inflated/deflated one by one.  otherwise the receiver might have to allocate a large chunk just to read an update for the latest slot.